### DISCUSSION TITLE
[Valheim] world backup

### DISCUSSION THREAD

**Discussion by the-archfiend06** [GameServerManagers/LinuxGSM#3353] [Q&A] [answered]
Hey all,
So I made a Valheim server on ubuntu and played on "world1" for a while until my hdd died. SO I switched to centos cause I had a clean install anyways. reinstalled lgsm and the server is up and running running "world2". If i wanted to run "world1" that I had in a backup (hidden files and all) how do you tell lgsm to load a different world? I have tried _default.cfg but it doesn't show a world load path that I can see.
Any help would be appreciated. I am getting balder by the minute XD

**Comment by EugeneGGHQ** [comment, marked as answer]
Assuming that all directories match the default ones, you would take advantage of the `gameworld` variable in your gameserver.cfg

`/home/vhserver/lgsm/config-lgsm/vhserver/vhserver.cfg` 

If you followed the documentation's default instructions that would be the path above, and the value would be:
"${selfname}"

In theory (though I haven't tested it) you could rename this value to world1 or world2 before relaunching the server to switch between save files.

If you try this, do let me know if it answers your question. **I suggest taking plenty of backups of your world folders as a precaution before doing any actual testing with your saves!**

**Reply by the-archfiend06** [reply]
So I made a backup copy of _default.cfg , changed the values over to both common.cfg & vhserver.cfg one at a time with different server names slightly so i could tell the difference and changed the world the backup world. It changes every other setting servername, password, etc., but it opens vhworld every time. So I deleted the vhworld files and added my world, renamed the variable in the cfg and it still loads the vhworld. 

We have a plan B by chance lol? I am going to keep looking around the server. I'll keep you posted if I find anything and thank you for the help so far.

EDIT 1: Removed extra Qquote text

EDIT 2: Installed Valheim server just with steamcmd on centos and it still loads the first world. so I was curious if my backup was even right and when i put it in my game files on windows loads the proper world. now i am very confused XD.

Below is the server details (while the server is off)
`Distro Details
=================================
Distro:   CentOS Linux 7 (Core)
Arch:   x86_64
Kernel:   3.10.0-1160.15.2.el7.x86_64
Hostname:   trevor-PowerEdge-T310
Uptime:   0d, 17h, 0m
tmux:   tmux 1.8
glibc:   2.17

Server Resource
=================================
CPU   
Model:   Intel(R) Xeon(R) CPU           X3440  @ 2.53GHz
Cores:   8
Frequency:   2526.969MHz
Avg Load:   0.01, 0.07, 0.05

Memory   
Mem:   total   used   free   cached  available
Physical:   7.7GB   424MB  7.0GB  5.5GB   7.0GB
Swap:   7.9GB   1.6MB  7.9GB

Storage
Filesystem:   /dev/mapper/centos-home
Total:     1022G
Used:     5.2G
Available:   1017G

Network
IP:   0.0.0.0
Internet IP:   216.174.84.121

Game Server Resource Usage
=================================
CPU Used:   0%
Mem Used:   0%   0MB

Storage
Total:   5.2G
Serverfiles:   1004M
Backups:   2.9G

Valheim Server Details
=================================
Server name:   Prepare To Suffer!!
App ID:   896660
Server IP:   0.0.0.0:2456
Internet IP:   216.174.84.121:2456
Server password:   ********
Game world:   suffering
Status:   OFFLINE

vhserver Script Details
=================================
Script name:   vhserver
LinuxGSM version:   v21.1.3
glibc required:   2.15
Discord alert:   off
Email alert:   off
IFTTT alert:   off
Mailgun (email) alert:   off
Pushbullet alert:   off
Pushover alert:   off
Rocketchat alert:   off
Slack alert:   off
Telegram alert:   off
Update on start:   off
User:   lgsm
Location:   /home/lgsm

Backups
=================================
No. of backups:   2
Latest backup:
    date:   Wed Mar 10 16:39:57 CST 2021 (less than 1 day ago)
    file:   /home/lgsm/lgsm/backup/vhserver-2021-03-07-161021.tar.gz
    size:   756M

Command-line Parameters
=================================
 ./valheim_server.x86_64 -name 'Prepare To Suffer!!' -password ******* -port 2456 -world suffering -public 1

Ports
=================================
Change ports by editing the parameters in:
/home/lgsm/lgsm/config-lgsm/vhserver

Useful port diagnostic command:
netstat -atunp | grep valheim

DESCRIPTION          DIRECTION  PORT  PROTOCOL
> Game   INBOUND    2456  udp
> Query  INBOUND    2457  udp'

**Reply by EugeneGGHQ** [reply]
Hey! Apologies, I think I misread your initial question, that's 100% my fault. 

If it's your wish to restore the files from your world1 save, you should just need to plop them in:

`~/.config/unity3d/IronGate/Valheim `

This directory is hidden (as inferred by the leading full-stop in .config) so it's not terribly obvious.

Let me know if this solves - all the other paths and variables can remain as default.

**Reply by the-archfiend06** [reply]
Hey. No worries.

I put the files in there. it still wasn't working. I found out the issue was actually in the _default.cfg that was downloaded. it needed quotes around every variable except for the port, then it worked like a charm.

You're first post was technically correct XD.
Thanks for the help.
